.github/CONTRIBUTING.md

```diff
@@ -1,2 +1,39 @@
-Thanks for contributing. If creating a pull request please make sure to submit it to the `devel` branch and not to `master`.
-Happy coding!
+### Thanks for contributing!
+
+# Branch
+Create the pull request against the `devel` branch instead of the `master`.
+
+# Quality
+All pull requests are welcome regardless of quality. We will work together to review and improve the pull request.
+
+Of course there are some steps that could be considered to improve the quality of the pull request and to support the process of review and development.
+
+- Make sure your code follows [PEP 8 — Style Guide for Python Code](https://pep8.org/). You can use some [linting tools](https://en.wikipedia.org/wiki/Lint_(software)) to automate the process (e.g. [`pycodestyle`](https://pycodestyle.pycqa.org) or [`flake8`](https://github.com/pycqa/flake8)).
+- Test your code using the existing test cases. For details read further.
+- It would be great if you could create [`unittest`](https://docs.python.org/3/library/unittest.html)'s for your code.
+
+# Unittesting
+
+This project uses pythons default [`unittest`](https://docs.python.org/3/library/unittest.html) package. You can run all test cases via the *discover* feature when you run this in the projects root folder.
+
+```sh
+python3 -m unittest discover
+```
+
+The output should look like this
+```sh
+----------------------------------------------------------------------
+Ran 74 tests in 0.823s
+
+OK
+```
+
+If you don't want to run all tests at once but a specific test case or method you need to *install* the package first. It is possible via a virtual environment but we recommend to use the `--editable` switch of `pip`. Run this in the projects root folder:
```

Comment:
```suggestion
If you want to run a specific test case or method you need to *install* the package first. It is recommended to use a virtual environment and the  `--editable` flag of `pip`. Run this in the projects root folder:
```

Reply:
It is the first time I use the review feature of GitHub. Is this the right way doing thins? ;)

You suggest "virtual environment and the".
But I meant "virtual environment or the".

Reply:
I know, I deliberately used "and the". These are not mutually exclusive, and not using a virtual environment will prevent one from using both a public/released version of the library and the development (local) version.